src/redisraft.c

```diff
@@ -697,42 +701,36 @@ static void redirectCommand(RedisRaftCtx *rr, RedisModuleCtx *ctx, Node *leader)
     replyRedirect(ctx, 0, &leader->addr);
 }
 
-static RRStatus handleNonLeaderCommand(RedisRaftCtx *rr, RedisModuleCtx *ctx, Node *leader, RaftRedisCommandArray *cmds)
+static void handleNonLeaderCommand(RedisRaftCtx *rr, RedisModuleCtx *ctx, Node *leader, RaftRedisCommandArray *cmds)
 {
     RedisModule_Assert(!raft_is_leader(rr->raft));
     RedisModule_Assert(cmds);
 
-    /* reply ASK for ASKING state commands */
-    if (cmds->asking) {
+
+    if (!leader) {
```

Comment:
Going back to same discussion :) I'd prefer previous version. I find it harder to understand when there are bunch of if checks. It's common to have "fallthrough" if checks for error cases/conditions and I find it easier to read:

```c
if (!leader) {
    .....
    return; 
}

if (rr->config->follower_proxy) {
    /* forward commands to the leader in follower_proxy state */
   .......
    return;
}

if (asking) {
    ......
    return;
}

redirect();
return;
```

Reply:
ok. redid